build/initialize-pipeline.ps1

```diff
@@ -9,4 +14,32 @@ if ($buildReason -eq "PullRequest") {
     Write-Host "##vso[task.setvariable variable=BuildArtifacts;isOutput=true]true"
     Write-Host "Setting 'BuildArtifacts' to true."
   }
-}
\ No newline at end of file
+}
+
+# Get major, minor and patchVersions
+[xml]$XMLContents = [xml](Get-Content -Path ".\build\common.props")
+$XMLContents.GetElementsByTagName("MajorVersion") |  ForEach-Object {
+  $majorVersion = $_.InnerText
+  Write-Host "##vso[task.setvariable variable=MajorVersion;isOutput=true]$majorVersion"
+  Write-Host "Setting 'MajorVersion' to $majorVersion"
+}
+
+$XMLContents.GetElementsByTagName("MinorVersion") |  ForEach-Object {
+  $minorVersion = $_.InnerText
+  Write-Host "##vso[task.setvariable variable=MinorVersion;isOutput=true]$minorVersion"
+  Write-Host "Setting 'MinorVersion' to $minorVersion"
+}
+
+$XMLContents.GetElementsByTagName("PatchVersion") |  ForEach-Object {
+  $patchVersion = $_.InnerText
+  Write-Host "##vso[task.setvariable variable=PatchVersion;isOutput=true]$patchVersion"
+  Write-Host "Setting 'PatchVersion' to $patchVersion"
+}
+
+#Update buildnumber with the same (Will be used by release pipelines)
+$customBuildNumber = "$majorVersion.$minorVersion.$patchVersion"
+if(($buildReason -eq "PullRequest") -or !($sourceBranch.ToLower().Contains("release")))
+{
+  $customBuildNumber = "$customBuildNumber-$buildNumber"
+}
+Write-Host "##vso[build.updatebuildnumber]$customBuildNumber"
```

Comment:
@brettsam FYI. It's weird that this didn't work before (iteration 3) but it's working now.